tools/slang-capability-generator/capability-generator-main.cpp

```diff
@@ -259,11 +421,82 @@ struct CapabilityConjunction
     }
 };
 
+String tryGetCapabilityName(const CapabilityDef* def)
+{
+    if(def)
+        return def->name;
+    return "[none]";
+}
+
 struct CapabilityDisjunction
 {
     List<CapabilityConjunction> conjunctions;
 
-    void addConjunction(const CapabilityConjunction& c)
+    void addConjunction(DiagnosticSink* sink, SourceLoc sourceLoc, const CapabilityConjunction& c)
+    {
+        if (c.isImpossible())
+            return;
+        bool cImpliesThis = false;
+        for (Index i = 0; i < conjunctions.getCount();)
+        {
+            // implied sets will be replaced
+            if (c.implies(conjunctions[i]))
+            {
+                cImpliesThis = true;
+                conjunctions.fastRemoveAt(i);
+            }
+            else
+                i++;
+        }
+        if (cImpliesThis)
+        {
+            conjunctions.add(_Move(c));
```

Comment:
This `_Move` will not work as you may expect.
The parameter, `c`, must be an rvalue-reference, "&&".

Reply:
Again, performance shouldn't matter too much here in this generator code that runs only once during build.